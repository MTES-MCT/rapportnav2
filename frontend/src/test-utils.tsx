import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
