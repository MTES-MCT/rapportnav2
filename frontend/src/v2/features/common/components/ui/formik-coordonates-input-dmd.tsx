import { CoordinatesFormat, FormikCoordinatesInput, FormikCoordinatesInputProps } from '@mtes-mct/monitor-ui'
import styled from 'styled-components'

export const FormikCoordinateInputDMD = styled(
  ({ label, ...props }: Omit<FormikCoordinatesInputProps, 'coordinatesFormat'>) => (
    <FormikCoordinatesInput
      label={`${label} (01° 12.600′ S 044° 58.800′ E)`}
      coordinatesFormat={CoordinatesFormat.DEGREES_MINUTES_DECIMALS}
      {...props}
    />
  )
)({})
