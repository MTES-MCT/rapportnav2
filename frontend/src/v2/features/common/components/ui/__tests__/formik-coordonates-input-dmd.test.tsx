import { Formik } from 'formik'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { FormikCoordinateInputDMD } from '../formik-coordonates-input-dmd'

const handleSubmit = vi.fn()

describe('FormikCoordinateInputDMD', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <Formik initialValues={{ coords: [] }} onSubmit={handleSubmit}>
        <FormikCoordinateInputDMD name={'coords'} label="my coords" />
      </Formik>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
