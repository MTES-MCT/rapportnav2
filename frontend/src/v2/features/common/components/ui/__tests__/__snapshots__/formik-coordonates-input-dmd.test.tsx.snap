// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormikCoordinateInputDMD > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        class="sc-fEOKFM eNpmbc Element-Fieldset sc-cHtFsA fyBNgP Field-CoordinatesInput sc-btRGtK"
      >
        <legend
          class="sc-beKSRx bPgRzG Element-Legend"
        >
          my coords (12° 12.12′ N 121° 21.21′ E)
        </legend>
        <div
          class="sc-bxmmIP bLqgCv Element-Fieldset__InnerBox"
        >
          <div
            class="sc-euHrWb fcMOjf"
          >
            <input
              data-cy="dmd-coordinates-input"
              name="coords"
              placeholder="__° __.___′ _ ___° __.___′"
              value=""
            />
            <span
              class="sc-ienWRC iSzZX"
            >
              (DMD)
            </span>
            <span
              class="sc-jJhUpn dKKttC"
            />
          </div>
        </div>
      </fieldset>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <fieldset
      class="sc-fEOKFM eNpmbc Element-Fieldset sc-cHtFsA fyBNgP Field-CoordinatesInput sc-btRGtK"
    >
      <legend
        class="sc-beKSRx bPgRzG Element-Legend"
      >
        my coords (12° 12.12′ N 121° 21.21′ E)
      </legend>
      <div
        class="sc-bxmmIP bLqgCv Element-Fieldset__InnerBox"
      >
        <div
          class="sc-euHrWb fcMOjf"
        >
          <input
            data-cy="dmd-coordinates-input"
            name="coords"
            placeholder="__° __.___′ _ ___° __.___′"
            value=""
          />
          <span
            class="sc-ienWRC iSzZX"
          >
            (DMD)
          </span>
          <span
            class="sc-jJhUpn dKKttC"
          />
        </div>
      </div>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
